core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3917,6 +3918,38 @@ public static List mapValues(Map map) {
     return new ArrayList<>(map.values());
   }
 
+  /** Support the MAP_FROM_ARRAYS function. */
+  public static Map mapFromArrays(List keysArray, List valuesArray) {
+    if (keysArray.size() != valuesArray.size()) {
+      throw new IllegalArgumentException(
```

Comment:
can this exception be added as a constant to Resources like the other errors thrown from this class?

Reply:
use Resources